project-base/app/src/Model/Product/Flag/FlagRepository.php

```diff
@@ -63,7 +63,13 @@ public function getVisibleFlagById(int $flagId, string $locale): Flag
             ->setParameter('flagId', $flagId)
             ->setParameter('locale', $locale);
 
-        return $flagsQueryBuilder->getQuery()->getSingleResult();
+        $flag = $flagsQueryBuilder->getQuery()->getOneOrNullResult();
+
+        if ($flag === null) {
+            throw new FlagNotFoundException(sprintf('Flag with ID "%s" does not exist.', $flagId));
```

Comment:
can it still be a problem in flagBreadcrumbGenerator? The exception is not caught there...